### ISSUE TITLE
Integer division of ActiveSupport::Duration instance returns zero

### ISSUE THREAD

**Issue by aarongough** [rails/rails#33179]
### Steps to reproduce

```shell
~ rails new division_test
[ ... snip 202 lines ... ]
~ cd division_test
~ rails console

Running via Spring preloader in process 13824
Loading development environment (Rails 5.2.0)
2.3.1 :001 > 1.year / 2
 => 0 seconds
```

### Expected behavior
In previous versions of Rails this behaved reasonably and would return a new `ActiveSupport::Duration` object representing half the original duration.

The expected behaviour is still currently seen with floats, it only behaves incorrectly with integers:

```ruby
1.year / 2
=> 0 seconds

1.year / 2.0
=> 0.5 years
```

### System configuration
**Rails version**: Rails 5.2.0
**Ruby version**: ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]

**Comment by aarongough** [comment]
Following up on this, further strange behaviour is exhibited when working with the results of a division stemming from this:

```ruby
> 1.day / 24
=> 0 seconds

> 3600 == (1.day / 24)
=> true

> Time.new(2018, 1, 1, 0, 0, 0, "-00:00") - (1.day / 24)
=> 2018-01-01 00:00:00 +0000
# should be 2017-12-31 23:00:00 +0000
```

**Comment by aarongough** [comment]
Upon further investigation it looks like the `#parts` of the Duration object resulting from the division are not being set correctly:

```ruby
> (1.day / 10).parts
=> {}

> (1.day / 10.0).parts
=> {:days=>0.1}

> (1.day / 10).value
=> 8640

> (1.day / 10.0).value
=> 8640.0
```

**Comment by percysnoodle** [comment]
Seen again in Rails 6:
```
$ bin/rails --version
Rails 6.0.2.1
$ bin/rails console
> 1.hour / 2
=> 0 seconds
```

**Comment by skipkayhil** [comment]
The inspect issue appears to have been fixed by #37839, and the `Time.new(...) - (1.day / 24)` issue was fixed in #37849

I believe the difference between `#parts` for `1.day / 10` and `1.day / 10.0` is intentional because `1 / 10` would be a loss of precision for the `days` part (but `1 / 10.0` does not lose precision)
